### PR TITLE
Added Tests to the Leaderboard Use Case

### PR DESCRIPTION
**Pull request by benmarlow**
Just cleaned up a few things in aligning the rows of the leaderboard.

### CODE REVIEW

File: src/test/java/LeaderboardTest/LeaderboardTest.java [hunk 1]
```diff
@@ -24,6 +53,7 @@ public void testGetAllUsersNonEmpty() {
     }
     @Test
     public void testLeaderboardTwoUsers() {
+        Leaderboard newBoard = interactor.updateLeaderboard();
```
Comment (eladperlman) [inline src/test/java/LeaderboardTest/LeaderboardTest.java:56]: is this test complete?

Comment (inaam10-tech) [reply src/test/java/LeaderboardTest/LeaderboardTest.java:56]: Yeah everything else seems good except for the incomplete test

File: src/main/java/LeaderboardUseCase/LeaderboardResponse.java [hunk 2]
```diff
@@ -23,8 +23,12 @@ public List<String> toStringList() {
         List<String> result = new ArrayList<>();
         int position = 1;
         for (User u : board.getTopUsers().keySet()) {
-            result.add(position + ". " + u.getUsername() + ":             $" + board.getTopUsers().get(u));
-            position++;
+            if(u.getUsername().length() < 25) {
+                result.add(position + ". " + u.getUsername() + ":" + new String(new char[20 - u.getUsername().length()]).replace("\0", " ") + "$" + board.getTopUsers().get(u));
+            } else {
+                result.add(position + ". " + u.getUsername() + ":           $" + board.getTopUsers().get(u));
+            }
```
Comment (inaam10-tech) [inline src/main/java/LeaderboardUseCase/LeaderboardResponse.java:30]: try refactoring code

**Review by inaam10-tech — DISMISSED** [review]
Seems good. Be sure to add tests later

**Review by eladperlman — APPROVED** [review]
Looks good!

**Review by Peaceful36 — APPROVED** [review]
The tests look good for me

**Review by Peaceful36 — APPROVED** [review]
The tests look good for me

**Review by eladperlman — APPROVED** [review]
Tests look complete, good job

File: src/main/java/LeaderboardUseCase/LeaderboardResponse.java (hunk 2 above)
Comment (benmarlow) [reply src/main/java/LeaderboardUseCase/LeaderboardResponse.java:28]: Yeah got this to work a bit better.

**Review by noel-toms — APPROVED** [review]
Nice tests, they seem to encompass everything that needs to be covered.

File: src/main/java/LeaderboardUseCase/LeaderboardResponse.java [hunk 3]
```diff
@@ -23,8 +23,12 @@ public List<String> toStringList() {
         List<String> result = new ArrayList<>();
         int position = 1;
         for (User u : board.getTopUsers().keySet()) {
-            result.add(position + ". " + u.getUsername() + ":             $" + board.getTopUsers().get(u));
-            position++;
+            if(u.getUsername().length() < 25) {
+                result.add(position + ". " + u.getUsername() + ":" + new String(new char[20 - u.getUsername().length()]).replace("\0", " ") + "$" + board.getTopUsers().get(u));
+            } else {
+                result.add(position + ". " + u.getUsername() + ":           $" + board.getTopUsers().get(u));
```
[the remark this answers is not in the document]
Comment (benmarlow) [reply src/main/java/LeaderboardUseCase/LeaderboardResponse.java:27]: Aligns everything with the prices.

**Pull request merged**